low-level/files/opt/seagate/sspl/bin/conf_based_sensors_enable

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e -u -o pipefail
+
+to_lower_case() {
+    tr '[:upper:]' '[:lower:]'
+}
+
+declare -a sed_expr=(-e '/^\[[A-Z]\+\]$/h')
+storage_type="$(provisioner get_setup_info --output json | jq -r .ret.storage_type | to_lower_case)"
+server_type="$(provisioner get_setup_info --output json | jq -r .ret.server_type | to_lower_case)"
+
+declare -A monitor=(
+    [REALSTORSENSORS]="true"
+    [NODEHWSENSOR]="true"
+    [SYSTEMDWATCHDOG]="true" # This is the 'server disks' sensor
+    [RAIDSENSOR]="true"
+    [SASPORTSENSOR]="true"
+    [MEMFAULTSENSOR]="true"
+    [CPUFAULTSENSOR]="true"
+
+)
+
+case "${storage_type}" in
+    virtual) 
+        monitor[REALSTORSENSORS]="false"
+        monitor[SYSTEMDWATCHDOG]="false"
+        ;;
+    jbod)
+        monitor[REALSTORSENSORS]="false"
+        ;;
+esac
+
+case "${server_type}" in
+    virtual)
+        monitor[NODEHWSENSOR]="false"
+        monitor[SASPORTSENSOR]="false"
+        monitor[MEMFAULTSENSOR]="false"
+        monitor[CPUFAULTSENSOR]="false"
+        ;;
+esac
+
+for group in "${!monitor[@]}"; do
+    # We remove comments after the value since those are improperly parsed by the ConfigReader
+    # in sspl_test.
+    sed_expr+=(-e '/^monitor=.*$/{ x ; /^\['"${group}"'\]$/{ x ; s/^monitor=.*$/monitor='"${monitor[${group}]}"'/;  b} ; x }')
```

Comment:
btw config reader class also has interface to write back in same config file. So if you see this code creating problem, going pythonic way is definitely an option to consider and using those config reader interfaces.
check https://github.com/Seagate/cortx-monitor/pull/240/files#
salt_util.py -> def update_config_file() line 163

Reply:
The sed command seems to be working so far.
But yes, if this get more complicated in the future we should consider moving to python, though that will have its own issues ( esp. calling into shell scripts from python and vice-versa during init. It may be required to change the entire file to python).

Reply:
We can one have one simple python script which would just update required attribute in given section, with section/attribute names as cmd line arguments to the script. that way shell <-> python switch wont be felt that much

Reply:
We will go ahead with SED approach as @pranav-seagate is confident its working fine.